sub_peers.js

```diff
@@ -27,13 +28,67 @@ var PeerHeap = require('./peer_heap.js');
 
 function TChannelSubPeers(channel, options) {
     TChannelPeersBase.call(this, channel, options);
+
+    var self = this;
     this.peerScoreThreshold = this.options.peerScoreThreshold || 0;
     this._heap = new PeerHeap(channel.random);
     this.choosePeerWithHeap = channel.choosePeerWithHeap;
+
+    this.hasMinConnections = typeof options.minConnections === 'number';
+    this.minConnections = options.minConnections;
+
+    this.currentConnectedPeers = 0;
+
+    this.boundOnOutConnectionIncrement = boundOnOutConnectionIncrement;
+    this.boundOnOutConnectionDecrement = boundOnOutConnectionDecrement;
+
+    function boundOnOutConnectionDecrement(_, peer) {
+        self.onOutConnectionDecrement(peer);
+    }
+    function boundOnOutConnectionIncrement(_, peer) {
```

Comment:
seems maybe a missed opportunity to transmit a delta number in the event, and replace two events with one?